### PR TITLE
Add pyqtgraph recipe

### PR DESCRIPTION
**Pull request by astaric**

### CODE REVIEW

File: recipes/pyqtgraph/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
```
Comment (astaric) [inline recipes/pyqtgraph/meta.yaml]: See the next build requirement (in the next line) :)

**Comment by astaric** [comment]
I have followed the steps on gitter to enable the CircleCI (followed my fork).

I have removed numpy x.x from build requirements and x.x part from install requirements. Regarding the versions, please let me know if you find out anything. 

Should I add you as a maintainer since you are a contributor of the base library (and apparently active on conda-forge)?

**Comment by ericdill** [comment]
> I have removed numpy x.x from build requirements and x.x part from install requirements. Regarding the versions, please let me know if you find out anything.

Will do.  If I don't hear back from Luke, we can just give this our best effort regarding pairings of pyqt4/5 and py2.7/3.4/3.5 and fix the recipe if/when bug reports start coming in

> Should I add you as a maintainer since you are a contributor of the base library (and apparently active on conda-forge)?

Sure thing. Thanks.

**Comment by campagnola** [comment]
On python 2.7 we can use pyqt4 or pyside; on python 3.x we can use pyqt4, pyqt5, or pyside. As far as I know, there are no new issues from 3.4 to 3.5. Ideally, the user should be able to install any combination of these qt libraries, since each application may have different requirements. Last I checked, there was no way to have pyqtgraph depend on `pyqt | pyside`. If that's still the case, then the pyqt dependency should probably be left exactly as it is.

**Comment by astaric** [comment]
I have updated the requirements to require pyqt4 on py2k and pyqt otherwise. The linux and osx builds still fail, but with a different error (https://circleci.com/gh/astaric/staged-recipes/tree/pyqtraph).

[Build #3](https://circleci.com/gh/astaric/staged-recipes/3) installed the following packages in the testing phase:

```
The following NEW packages will be INSTALLED:

    numpy:     1.11.2-py27_blas_openblas_202 conda-forge                            [blas_openblas]
    pyqtgraph: 0.10.0-np111py27_0            file:///staged-recipes/build_artefacts
````

And failed with missing python error

[Build #5] (https://circleci.com/gh/astaric/staged-recipes/5) installed

```
The following NEW packages will be INSTALLED:

    numpy:     1.11.2-py34_blas_openblas_202 conda-forge                            [blas_openblas]
    pyqt:      4.11.4-py34_1                 conda-forge                           
    pyqtgraph: 0.10.0-py34_0                 file:///staged-recipes/build_artefacts
    python:    3.4.5-2                       conda-forge                           
    sip:       4.18-py34_0                   conda-forge
```

and failed with missing qt libs error

[Build #8](https://circleci.com/gh/astaric/staged-recipes/8) installed

```
The following NEW packages will be INSTALLED:

    cairo:      1.14.6-0                      conda-forge                           
    fontconfig: 2.11.1-5                      conda-forge                           
    gettext:    0.19.7-1                      conda-forge                           
    glib:       2.51.0-1                      conda-forge                           
    harfbuzz:   1.0.6-3                       conda-forge                           
    jpeg:       9b-0                          conda-forge                           
    libffi:     3.2.1-2                       conda-forge                           
    libtiff:    4.0.6-7                       conda-forge                           
    numpy:      1.11.2-py34_blas_openblas_202 conda-forge                            [blas_openblas]
    pango:      1.40.3-0                      conda-forge                           
    pcre:       8.38-1                        conda-forge                           
    pixman:     0.34.0-0                      conda-forge                           
    pyqt:       4.11.4-py34_1                 conda-forge                           
    pyqtgraph:  0.10.0-py34_0                 file:///staged-recipes/build_artefacts
    python:     3.4.5-2                       conda-forge                           
    qt:         4.8.7-3                       conda-forge                           
    sip:        4.18-py34_0                   conda-forge          
``` 

And failed with missing libpng.

A couple more force pushes and everything will start working :P

**Comment by astaric** [comment]
I can reproduce the failure on my local machine (macOS) by running 
```
conda-build-all . --matrix-conditions "numpy >=1.10" "python >=2.7,<3|>=3.4"
```
with conda-build 1.21.14, which is used on circleci/travis.

If I update conda-build to the latest version, the build completes successfully.

Any ideas how to proceed?

**Comment by ericdill** [comment]
You can reproduce (and therefore debug) the circle-ci builds locally if you have docker installed by running `bash scripts/run_docker_build.sh`

I am not sure why you're getting libpng issues.
```
import: 'pyqtgraph'
Traceback (most recent call last):
  File "/staged-recipes/build_artefacts/test-tmp_dir/run_test.py", line 26, in <module>
    import pyqtgraph
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/pyqtgraph/__init__.py", line 13, in <module>
    from .Qt import QtGui
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/pyqtgraph/Qt.py", line 133, in <module>
    from PyQt4 import QtGui, QtCore, uic
ImportError: libpng16.so.16: cannot open shared object file: No such file or directory
TESTS FAILED: pyqtgraph-0.10.0-py34_0
```

Doing a quick search of conda-forge.github.io and staged-recipes shows a few issues that seem to address libpng issues, but it is not clear to me what the solution here is.  Any ideas @ocefpaf @jakirkham?

**Comment by lgpage** [comment]
This could be related to the similar issue I had with `nodejs` where is gets installed and linked in the first conda build (or test) env but does not get linked in any of the following conda build (or test) envs

**Comment by ocefpaf** [comment]
I will re-build `pyqt` and check this.

**Comment by astaric** [comment]
@lgpage how did you solve/workaround the issue? Is there anything I should try?

**Comment by astaric** [comment]
@ocefpaf I am not sure that this is a pyqt issue. If use conda-build 1.21.14 to build multiple versions of this package (py2, py3.4, py3.5), the tests of the second always fails. If I do all three, 3.4 fails, if select only 3.4 and 3.5, 3.5 will fail. If I update to the latest conda-build, everything builds (and tests) just fine.

**Comment by lgpage** [comment]
@astaric see the `build.sh` script from https://github.com/conda-forge/staged-recipes/pull/2065
I am really hoping there is a better workaround or fix.

**Comment by ocefpaf** [comment]
>  If I update to the latest conda-build, everything builds (and tests) just fine.

Good to know. Maybe you should build this one with `conda-build 2` already and we can revert the change (or not :wink:  @jjhelmus) in `staged-recipes` later.

File: recipes/pyqtgraph/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - setuptools
+  run:
+    - python
+    - numpy
+    - pyqt 4.11.*  # [py2k]
+    - pyqt  # [not py2k]
```
Comment (jakirkham) [inline recipes/pyqtgraph/meta.yaml]: Why this pinning difference?

**Comment by campagnola** [comment]
I have a correction to make: for a long time we never tested Python2.7+PyQt5 simply because PyQt5 was only available for Python3+. I tested this combination today and everything seems to work, so I'd recommend relaxing the pyqt requirement.

**Comment by astaric** [comment]
good to know, I'll update the recipe

**Comment by scopatz** [comment]
Is this ready to go?

**Comment by astaric** [comment]
It is, I have recently rebased onto the latest master and the tests seem to pass now.

**Pull request merged**